pkg/syncer/k8scloudoperator/k8scloudoperator.go

```diff
@@ -123,13 +123,21 @@ func (k8sCloudOperator *k8sCloudOperator) GetPodVMUUIDAnnotation(ctx context.Con
 	if pv.Spec.ClaimRef == nil {
 		return nil, logger.LogNewErrorf(log, "No Claim ref found for this PV with volumeID: %s", volumeID)
 	}
-	podResult, err := k8sCloudOperator.getPod(ctx, pv.Spec.ClaimRef.Name, pv.Spec.ClaimRef.Namespace, nodeName)
-	if err != nil {
-		return nil, err
-	}
-	podName := podResult.Name
-	podNamespace := podResult.Namespace
+
+	var podName string
+	var podNamespace string
 	err = wait.Poll(pollTime, timeout, func() (bool, error) {
+		// Retrieve the pod name and namespace only if there are non-empty.
+		// If they are already pre-populated, use them to get the pod
+		// info from API server.
+		if len(podName) == 0 || len(podNamespace) == 0 {
+			podResult, err := k8sCloudOperator.getPod(ctx, pv.Spec.ClaimRef.Name, pv.Spec.ClaimRef.Namespace, nodeName)
+			if err != nil {
+				return false, err
+			}
+			podName = podResult.Name
+			podNamespace = podResult.Namespace
+		}
 		var exists bool
 		pod, err := k8sCloudOperator.k8sClient.CoreV1().Pods(podNamespace).Get(ctx, podName, metav1.GetOptions{})
```

Comment:
since we are now retrieving the pod in the `wait.Poll`, we can enhance this logic for the case, when the pod is retrieved with the annotation already set.
podResult might have the annotation with the value, we are retrieving pod again here.

Reply:
+1